src/app/outbound/mod.rs

```diff
@@ -147,15 +148,15 @@ where
     // over all endpoints returned from the destination service.
     let balancer_layer = svc::layers()
         .push_spawn_ready()
-        .push(resolve::layer(discovery::Resolve::new(resolve)))
+        .push(discover::Layer::new(2, resolve))
```

Comment:
nit: reading this, the meaning of the `2` is not super obvious at all --- WDYT about making it a constant, or adding a builder-type API to `discover::Layer` so it's clearer what 2 means?